main.js

```diff
@@ -241,87 +284,56 @@ function createPopup(index) {
+      });
+    })(i);
   }
+});
 
-  const projectCardButton = document.createElement('button');
-  projectCardButton.classList.add('project_card_button');
-  projectCardButton.setAttribute('type', 'button');
-  projectCardButton.innerText = 'See Project';
-  projectCardWrapper.appendChild(projectCardButton);
+// Form Validation //
 
-  projectWrapper.appendChild(projectCard);
+/* ------- SAVE INPUT VALUES TO LOCAL STORAGE -------*/
+const contactForm = document.querySelector('.contact__form');
 
-  projectCardButton.addEventListener('click', () => createPopup(i));
-}
+const saveFormData = () => {
+  const formData = {
+    fullName: document.getElementById('fullname').value,
+    email: document.getElementById('email').value,
+    message: document.getElementById('message').value,
+  };
 
-// Form Validation //
+  localStorage.setItem('formData', JSON.stringify(formData));
+};
 
-const fullName = document.getElementById('fullname');
-const email = document.getElementById('email');
+/* ------- GET INPUT VALUES FROM LOCAL STORAGE -------*/
+const loadFormData = () => {
+  const storedFormData = localStorage.getItem('formData');
 
-fullName.removeAttribute('required');
+  if (storedFormData !== null) {
+    const formData = JSON.parse(storedFormData);
+    document.getElementById('fullname').value = formData.fullName;
+    document.getElementById('email').value = formData.email;
+    document.getElementById('message').value = formData.message;
+  }
+};
 
-function errorMessage() {
-  const errorWindow = document.createElement('div');
-  errorWindow.setAttribute('id', 'error-window');
-  const errorMsg = document.createElement('div');
-  errorMsg.setAttribute('id', 'error-message');
-  errorMsg.setAttribute('class', 'error-message');
-  const errorMsgText = document.createTextNode(
-    'Your email should have only lowercase letters',
-  );
-  errorMsg.appendChild(errorMsgText);
-  errorWindow.appendChild(errorMsg);
-  email.after(errorWindow);
-}
+/* ------- SAVE DATA ON SUBMIT -------*/
 
-email.addEventListener('input', () => {
-  const emailValue = email.value;
-  const contactForm = document.getElementsByClassName('contact__form')[0];
-  const emailError = document.getElementById('error-message');
-  if (emailValue !== emailValue.toLowerCase()) {
-    if (contactForm.contains(emailError) === false) {
-      errorMessage();
-    }
-  } else if (document.body.contains(emailError) === true) {
-    emailError.remove();
-  }
+contactForm.addEventListener('submit', (e) => {
+  e.preventDefault();
+  saveFormData();
 });
 
-document
-  .getElementsByClassName('contact__form')[0]
-  .addEventListener('submit', (event) => {
-    const emailValue = document.getElementById('email').value;
-    if (emailValue !== emailValue.toLowerCase()) {
-      event.preventDefault();
-    }
-  });
\ No newline at end of file
+/* ------- LOAD DATA ON PAGE LOAD -------*/
+
+window.addEventListener('load', () => {
+  loadFormData();
+});
```

Comment:
- [ ] As stated by the previous reviewer, kindly make sure the data persists after reloading, and the data is saved in local storage. Presently, that's not the case as all data entered is gone after the page reload and it's not saved in local storage 


- In order to be saved to the local storage the data object is turned into a string using `JSON.stringify` or similar methods. Ie:

```
localStorage.setItem('contactFormInput', JSON.stringify(formData));

```

### View:
![image](https://github.com/mateoburvi/Portfolio/assets/50990925/6ceb875f-b4e2-4f85-8b13-994bdca15db0)

### Requirements:
![image](https://github.com/mateoburvi/Portfolio/assets/50990925/559d4985-9677-461c-acc6-b4ce1a690cc9)

Reply:
Please let me know if there are any further changes required, as I made all the changes you suggested, thanks a lot for your help, and sorry if I had any mistakes changing the things you are asking me to